include/boost/numeric/ublas/tensor/dynamic_tensor.hpp

```diff
@@ -0,0 +1,233 @@
+                "boost::numeric::ublas::dynamic_tensor( const OtherTensor& ) : The tensor should have same value type"
+            );
+
+            std::copy(other.begin(), other.end(), super_type::begin());
+        }
+
+        template<class OtherTensor>
+        constexpr dynamic_tensor (basic_tensor<OtherTensor> && other)
+            : super_type(other.extents())
+        {
+            static_assert( is_valid_tensor_v<OtherTensor>,
+                "boost::numeric::ublas::dynamic_tensor( OtherTensor&& ) : The OtherTensor should have a tensor type"
+            );
+
+            static_assert( std::is_same_v<value_type, typename OtherTensor::value_type>,
+                "boost::numeric::ublas::dynamic_tensor( OtherTensor&& ) : The tensor should have same value type"
+            );
+
+            std::move(other.begin(), other.end(), super_type::begin());
+
+        }
+
+        inline 
+        constexpr dynamic_tensor& operator=(const_reference v) noexcept{
+            std::fill(super_type::begin(), super_type::end(), v);
+            return *this;
+        }
+
+    };
+
+    template<typename V>
+    dynamic_tensor(dynamic_extents<> const&, V const&) -> dynamic_tensor<V,first_order>;
+
+    dynamic_tensor(dynamic_extents<> const&) -> dynamic_tensor<float,first_order>;
+
+    template<typename T>
+    dynamic_tensor(basic_tensor<T> const&) -> dynamic_tensor< typename T::value_type, typename T::layout_type >;
+
+    template<typename T>
+    dynamic_tensor(basic_tensor<T> &&) -> dynamic_tensor< typename T::value_type, typename T::layout_type >;
+
+} // boost::numeric::ublas
+
+
+namespace boost::numeric::ublas{
+
+    template<typename T, typename F>
+    struct tensor_traits< dynamic_tensor<T,F> > {
+        using container_type= std::vector< T >;
+        using extents_type  = dynamic_extents<>;
+        using layout_type   = F;
+        using container_tag = dynamic_tensor_tag;
+    };
+
+    template<typename T, typename F, typename NewValue>
+    struct tensor_rebind< dynamic_tensor<T, F>, NewValue > {
+        using type = dynamic_tensor< NewValue, F >;
+    };
+
+    template<typename T, typename F, typename NewValue, typename NewLayout>
+    struct tensor_rebind< dynamic_tensor<T, F>, NewValue, NewLayout > {
+        using type = dynamic_tensor< NewValue, NewLayout >;
+    };
+
+    template<typename T, typename F>
+    struct is_static< dynamic_tensor<T, F> > : std::false_type{};
+
+    template<typename T, typename F>
+    struct is_static_rank< dynamic_tensor<T, F> > : std::false_type{};
+
+    template<typename T, typename F>
+    struct is_dynamic< dynamic_tensor<T, F> > : std::true_type{};
+
+    template<typename T, typename F>
+    struct is_dynamic_rank< dynamic_tensor<T, F> > : std::true_type{};
+
+    template<typename V, typename F>
+    struct result_tensor< V, dynamic_extents<>, F >{
+        using type = dynamic_tensor< V, F >;
+    };
```

Comment:
Should be inside a `type_traits` header.

Reply:
will do once we have better type_traits structure. [Here](https://github.com/boostorg/ublas/pull/81#discussion_r418562362) is the question we need to answer.